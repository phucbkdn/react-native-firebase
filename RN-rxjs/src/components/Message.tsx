import React, { FC, memo } from 'react'
import { Avatar } from 'react-native-elements'
import { View, Text } from './Themed'
import { messageStyles } from './styles/message.styles'

interface MessageType {
  message: string
  time: string
  isPrimary?: boolean
  photoURL?: string
}

const Message: FC<MessageType> = memo(
  ({ message, time, isPrimary, photoURL }: MessageType) => (
    <View
      style={[
        isPrimary
          ? messageStyles.primaryWrapper
          : messageStyles.secondaryWrapper,
      ]}
    >
      <Avatar
        rounded
        size="small"
        source={{
          uri: photoURL,
        }}
      />
      <View
        style={[
          messageStyles.wrapper,
          isPrimary ? messageStyles.primary : messageStyles.secondary,
        ]}
      >
        <Text style={messageStyles.textStyle}>{message}</Text>
        <Text style={messageStyles.time}>{time}</Text>
      </View>
    </View>
  )
)

Message.defaultProps = {
  isPrimary: false,
  photoURL:
    'https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_1280.png',
}

export default Message
