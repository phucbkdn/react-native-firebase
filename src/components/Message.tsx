import React, { FC, memo } from 'react'
import { View, Text } from 'react-native'
import { messageStyles } from './styles/message.styles'

interface MessageType {
  message: string
  time: string
  isPrimary?: boolean
}

const Message: FC<MessageType> = memo(
  ({ message, time, isPrimary }: MessageType) => (
    <View
      style={[
        isPrimary
          ? messageStyles.primaryWrapper
          : messageStyles.secondaryWrapper,
      ]}
    >
      <View
        style={[
          messageStyles.wrapper,
          isPrimary ? messageStyles.primary : messageStyles.secondary,
        ]}
      >
        <Text style={messageStyles.textStyle}>{message}</Text>
        <Text style={messageStyles.time}>{time}</Text>
      </View>
    </View>
  )
)

Message.defaultProps = {
  isPrimary: false,
}

export default Message
