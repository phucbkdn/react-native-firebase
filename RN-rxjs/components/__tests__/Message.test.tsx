import React from 'react'
import { TouchableOpacity } from 'react-native'
import { render } from '@testing-library/react'
import  renderer from 'react-test-renderer'
import Message from '../Message'

const props = {
  name: 'Name',
  time: '1:00',
  isPrimary: true,
}

describe('Testing Message component', () => {
  test("Render correct component", () => {
    const { container } = render(<Message {...props} />)
    expect(container).toMatchSnapshot()
  })
})
