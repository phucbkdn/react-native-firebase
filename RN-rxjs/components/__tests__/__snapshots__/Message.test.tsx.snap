// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Message component Render correct component 1`] = `
<div>
  <view>
    <view>
      <text
        style="color: rgb(153, 153, 153); font-size: 18px; flex-wrap: wrap;"
      />
      <text
        style="color: rgb(147, 154, 164); font-size: 16px; text-transform: capitalize; padding-left: 20px;"
      >
        1:00
      </text>
    </view>
  </view>
</div>
`;
